components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -2208,6 +2347,20 @@ public static String getTenantDomainOfOauthApp(String clientId)
         return getTenantDomainOfOauthApp(oAuthAppDO);
     }
 
+    /**
+     * Get all the OAuth applications for the client ID.
+     *
+     * @param clientId Client ID.
+     * @return  Array of OAuthApp data objects.
+     * @throws IdentityOAuth2Exception      If an error occurred while retrieving the applications.
+     * @throws InvalidOAuthClientException  If an application not found for the given client ID.
+     */
+    public static OAuthAppDO[] getAllAppInformationByClientId(String clientId)
```

Comment:
Think of a better name (getAppsForConsumerKey or similar that's align with DAO method name)